### ISSUE TITLE
More liberal use of IdentifierExpressions in place of strings

### ISSUE THREAD

**Issue by bakkot** [shapesecurity/shift-spec#98]
It would be useful for the spec to have the property that any semantic static reference to a ES variable is represented as a `BindingIdentifier` or `IdentifierExpression`. This is currently _almost_ the case, but there's a couple of exceptions where strings are used instead:
- `ShorthandProperty`
- `name` attribute of `ExportSpecifier`

The first case is easily fixed: just change the `name` attribute of `ShorthandProperty` to be an `IdentifierExpression`.

The second case is more complicated, because the `name` attribute of an `ExportSpecifier` only references a variable if the `ExportSpecifier` in question appears in an `ExportFrom` without a `moduleSpecifier`. To resolve this (and because it makes sense anyway), I propose that `ExportFrom` and `ExportSpecifier` be split and have their interfaces changed. This might look as follows:

``` idl
interface ExportFrom : ExportDeclaration {
  attribute ExportFromSpecifier[] namedExports;
  attribute string moduleSpecifier;
};

interface ExportFromSpecifier : Node {
  attribute IdentifierName name;
  attribute IdentifierName? exportedName;
};

interface ExportNames : ExportDeclaration {
  attribute ExportNameSpecifier[] namedExports;
};

interface ExportNameSpecifier : Node {
  attribute IdentifierExpression name;
  attribute IdentifierName? exportedName;
};
```

Compare the current [ExportFrom](https://github.com/shapesecurity/shift-spec/blob/77c05b790de37d1f08f28bf00d6f8628a11f5114/spec.idl#L196-L199) and [ExportSpecifier](https://github.com/shapesecurity/shift-spec/blob/77c05b790de37d1f08f28bf00d6f8628a11f5114/spec.idl#L209-L212) definitions. Note in particular that the optional attribute in specifiers is now the `exportedName`, instead of the `name`: this is because `ExportNameSpecifier` necessarily references a variable, so its `name` attribute should not be optional. This is also symmetric with `ImportSpecifier`.

**Comment by michaelficarra** [comment]
:+1: Sounds great!

**Comment by bakkot** [comment]
Closed by #102.
